core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -189,7 +190,10 @@ protected ShardValidateQueryResponse shardOperation(ShardValidateQueryRequest re
             }
             if (request.rewrite()) {
                 explanation = getRewrittenQuery(searcher.searcher(), searchContext.query());
-            }   
+            }
+        } catch (QueryShardException e) {
+            valid = false;
+            error = e.getDetailedMessage();
         } catch (QueryParsingException e) {
```

Comment:
ok this was my bad...I misread the warning :) just replace the two catches with a single `catch(QueryShardException|QueryParsingException e)` ?